omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java

```diff
@@ -54,25 +70,68 @@ public String post(PageModel model, @BindParams UserDefaults userDefaults,
                     UiCommonsConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE, "adminui.account.defaults.fail");
             return "account/changeDefaults";
         }
-
-        model.addAttribute("userDefaults", userDefaults);
-        model.addAttribute("locales", Context.getAdministrationService().getPresentationLocales());
         return "myaccount/myAccount";
     }
 
-    public class UserDefaults {
+    public class LocaleOption {
 
-        private String defaultLocale;
-        private String proficientLocales;
+        private Locale locale;
+        private boolean selected;
+        private String name;
+        private String language;
 
-        public UserDefaults() {
+        public LocaleOption() {
```

Comment:
Do you really need this? Just label and code should be all you need where the code is the language or language and country